### PR TITLE
Jonmv/deployment risk

### PR DESCRIPTION
**Pull request by jonmv**
@freva or @mpolden please review. 

This is in preparation for slimming down the ApplicationVersion as a key, to just the RevisionId. Then, we can put more stuff in ApplicationVersion, such as commit messages and other metadata, and whether the user has decided to cancel a revision.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java [hunk 1]
```diff
@@ -422,19 +422,15 @@ public void finish(RunId id) throws TimeoutException {
                 logs.flush(id);
                 metric.jobFinished(run.id().job(), finishedRun.status());
 
-                DeploymentId deploymentId = new DeploymentId(unlockedRun.id().application(), unlockedRun.id().job().type().zone(controller.system()));
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java:421]: Ah, never mind this commit—I read that boolean inverted. This was correct after all. This was fixed later, when I rewrite this to also update the revision history.

**Review by mpolden — APPROVED** [review]
Nice cleanup!

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/RevisionHistory.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    public RevisionHistory with(ApplicationVersion revision) {
+        NavigableMap<RevisionId, ApplicationVersion> production = new TreeMap<>(this.production);
+        production.put(revision.id(), revision);
+        return new RevisionHistory(production, development);
+    }
+
+    /** Returns a copy of this with the new development revision added, and the previous version without a package. */
+    public RevisionHistory with(ApplicationVersion revision, JobId job) {
+        NavigableMap<JobId, NavigableMap<RevisionId, ApplicationVersion>> development = new TreeMap<>(this.development);
+        NavigableMap<RevisionId, ApplicationVersion> revisions = development.computeIfAbsent(job, __ -> new TreeMap<>());
```
Comment (freva) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/RevisionHistory.java:83]: By returning existing here we will modify the original instance

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/RevisionHistory.java:83]: Yes, thanks, forgot a copy here!

**Pull request merged**